content/en/blog/_posts/TBD-admission-controllers-for-container-drift/index.md

```diff
@@ -0,0 +1,90 @@
+
+```
+$ kubectl get pod --show-labels
+NAME      READY  STATUS   RESTARTS  AGE  LABELS
+test-pod  1/1    Running  0         2s   box.com/podInitialInteractionTimestamp=1632524400,box.com/podInteractorUsername=username-1,box.com/podTTLDuration=1h0m0s
+
+$ kubectl describe pod test-pod
+...
+Events:
+Type       Reason            Age     From                            Message
+----       ------            ----    ----                            -------
+Warning    PodInteraction    5s      admission-controller-service    Pod was interacted with 'kubectl exec' command by user 'username-1' initially at time 2021-09-24 16:00:00 -0800 PST
+Warning    PodInteraction    5s      admission-controller-service    Pod will be evicted at time 2021-09-24 17:00:00 -0800 PST (in about 1h0m0s).
+```
+
+### 3. Evict the target pod after a predefined period
+As you can see in the above event messages, the affected pod is not evicted immediately. At times, developers might have to get into their running containers necessarily for debugging some live issues. Therefore, we define a TTL of affected pods based on the environment of clusters they are running. In particular, we allow a longer time in our dev clusters as it is more common to run `kubectl exec` or other interactive commands for active development. While in prod clusters, there is a more limited time specified to avoid the affected pods serving traffic long term. The controller service internally sets a timer to each pod according to the given TTL duration. Once the timer is up, it will use the K8s Eviction API to restart the pod. The Eviction API ensures service availability since it respects the [PodDisruptionBudget](https://kubernetes.io/docs/concepts/workloads/pods/disruptions/) (PDB) corresponding to the Deployment object. Let's say if a user has defined X number of pods as critical in their PDB object, the eviction process, requested by our admission controller service, will not continue when the Deployment has fewer than X pods running.
+
+Here comes a sequence diagram of the entire workflow mentioned above: 
+{{< figure src="https://github.com/box/kube-exec-controller/raw/main/workflow-diagram.png" alt="Workflow Diagram" width="75%" >}}
```

Comment:
If you could make this a SVG diagram, that'd be champion. Check out https://mermaidjs.github.io/ for one way to do that.

Reply:
Also, change `width="75%"` to `class="diagram-medium"` or `class="diagram-large"`.

Reply:
It doesn't have to be SVG format but we do strongly prefer that.

Reply:
Updated to SVG format and changed the attribute as suggested.